src/android/com/mady/wifi/api/WifiStatus.java

```diff
@@ -284,13 +287,39 @@ public boolean pingCmd(String addr) {
             int exit = pro.exitValue();
             if (exit == 0) {
                 return true;
-            } else {
-                //ip address is not reachable
-                return false;
             }
         } catch (IOException e) {
+            Log.d("RAVEN", "ICMP ping error - " + e.getMessage());
+        }
+        Log.d("RAVEN", "ICMP ping to " + addr + " failed");
+        if (isReachable(addr, 443, 5000)) return true;
+        return isReachable(addr, 80, 5000);
```

Comment:
Doesn't your API require HTTPS? What's the value of checking if port 80 is reachable when 443 is not? (I'm assuming that `addr` is the Raven API)

Reply:
So `addr` is actually `www.google.com`. You're right that it probably doesn't make sense to check on port 80 since if port 443 failed, the users won't be able to connect to our API anyway.